src/scripts/components/CRCMaker.js

```diff
@@ -312,6 +325,18 @@ export default class CRCMaker extends Component {
           ) }
         </main>
 
+        { state.importVisible &&
+          <Dialog title='Import from JSON' onClose={this.toggleImport}>
+            <textarea id='json-import' onChange={e => this.inputJson = e.target.value}></textarea>
+
+            <button onClick={() => {
```

Comment:
Extract the logic into a function in the class. Also try to handle the case where the JSON might be improperly formatted.

Reply:
Yep good idea. I can put that in its own component.